simpleviewinc/keg-hub
# Zen 468 pressable update

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-468)

## Latest Updates (12/4/20)
* updated retheme pointerstate names
* added prop for button for selectable
* updated packages


## Context

* when we recently upgrade to rnw v14, it introduced a new way for
hover to work
* one approach is to use a new [Pressable](https://reactnative.dev/docs/pressable#children) component that passes the hover boolean to child components

## Goal
* Fix the bug where the styles are disappearing upon click+drag action
* update `keg-components` `Touchable` to use the new `Pressable` component from react-native
* ensure all components that uses `Touchable` still work the same as before
* Update the Touchable story

## Updates

### keg-components
* `repos/keg-components/src/components/button/button.js`
  * updated jsdoc
  * added some props that's relevant to the `Pressable` component

* `repos/keg-components/src/components/textToggle/textToggle.examples.js`
   * minor styling update for storybook

* `repos/keg-components/src/components/touchable/touchable.examples.js`
    * added more examples for Touchable component

* `repos/keg-components/src/components/touchable/touchable.native.js` 
    * updated to use `Pressable` component
    * by default, it utilizes the `onPressIn` and `onPressOut` to determine interaction state
    * users can still pass in functional child if they want full control over these states (`hovered, pressed, focused`)

* `repos/keg-components/src/components/touchable/touchable.stories.mdx`
    * added detail in the docs for Touchable

* `repos/keg-components/src/utils/helpers/getActiveOpacity.js`
   * updated so it works the way it was intended to

### re-theme
* `repos/re-theme/src/hooks/usePointerState.js`
   * updated pointer state to allow it to work on mobile devices 
       * ref https://patrickhlauke.github.io/touch/tests/results/

## Testing
### Re-theme
* Pull this branch
* Run `keg retheme && yarn test`
* Ensure all test passes

### Keg-components
* Run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:zen-468-pressable-update`
* navigate to http://components-zen-468-pressable-update.local.kegdev.xyz/?path=/docs/components-button-contained--standard
* ensure that you can't bug out any of the components by clicking + dragging
* Components to test
   * `Button`
       * ensure the hover and active styles are working

   * `Display/Icon/Touchable`
      * verify that you see opacity change when clicking on it
  
   * `Form/Input/onPress`
      * verify that an action is logged in the `Actions` tab when you click on it
 
   * `Interactions/textToggle`
      * ensure toggle works as expected and doesn't bug out

   * `Touchable`
      * `Basic` example should have an opacity change on press by default
      * `With States` example should change text to `hovered` when hovered. and `hovered and clicked` when pressed

### Tap Evf
* Run `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-468-pressable-update`
* navigate to http://evf-zen-468-pressable-update.local.kegdev.xyz/
* Verify that the affected components now work/look as intended
* The following components are affected
    * **Session Link** (session name/session presenters)
          * *Web*
               * it should change opacity when you click on it
               * it should NOT lose any styling when you click + drag
          * *mobile view*
               * it should not be blue colored
               * it should not be interactable (click should pass through and open/close the `Drawer` item

   * **Evf Buttons**
      * it should NOT lose any styling when you click + drag
      * it should have a `hover` style (opacity) when you hover over it and `active` (more opacity) style when you click on it
      * **Disabled** 
          * open `Filter` modal
          * select `machine learning` label
          * Verify that the Evf Button has the disabled style 

  * **Grid Item (Mobile View)**
     * Switch to Mobile view
     *  Switch to `Day 1`
     * Verify when you click the session grid item, it has an active opacity style
